### DISCUSSION TITLE
Future steps still running after previous ones failed

### DISCUSSION THREAD

**Discussion by mitchcurtis** [community/community#25681] [Actions]
<p>Hi.</p>
<p>The “Install Qt” step on <a href="https://github.com/mitchcurtis/slate/runs/3075356483" class="inline-onebox" rel="noopener nofollow ugc">Update packages.yml · mitchcurtis/slate@326d6d4 · GitHub</a> should fail due to install-qt.sh not existing, but it doesn’t, even though pipefail is apparently set by default:</p>
<p><a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#exit-codes-and-error-action-preference" class="onebox" target="_blank" rel="noopener nofollow ugc">https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#exit-codes-and-error-action-preference</a></p>
<p><a href="https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#job-status-check-functions" class="inline-onebox" rel="noopener nofollow ugc">Context and expression syntax for GitHub Actions - GitHub Docs</a> says:</p>
<blockquote>
<p>You can use the following status check functions as expressions in if conditionals. A default status check of success() is applied unless you include one of these functions.</p>
</blockquote>
<p>The docs for success say:</p>
<blockquote>
<p>Returns true when none of the previous steps have failed or been canceled.</p>
</blockquote>
<p>So why didn’t the Install Qt step fail?</p>
<p>Cheers.</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="mitchcurtis" data-post="1" data-topic="191164">
<div class="title">
<div class="quote-controls"></div>
 mitchcurtis:</div>
<blockquote>
<p>even though pipefail is apparently set by default</p>
</blockquote>
</aside>
<p>The log for the “Install Qt” step in your workflow says this:</p>
<pre><code class="lang-yaml">  shell: /bin/bash -e {0}
</code></pre>
<p>So I’m wondering if <code>pipefail</code> really is enabled, regardless of what the docs say. Maybe try adding <code>set -o pipefail</code> in the step and see if it makes a difference? And if it does I fear it’s time for a bug report. :sweat_smile:</p>

**Comment by mitchcurtis** [comment]
<p>Thanks, adding <code>set -o pipefail</code> before running the scripts made the rest fail:</p>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://github.com/mitchcurtis/slate/runs/3115384417">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">

      <a href="https://github.com/mitchcurtis/slate/runs/3115384417" target="_blank" rel="noopener nofollow ugc">GitHub</a>
  </header>

  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/6002340/181092478-45ac6c30-ebff-401e-bc03-ec645f993256.png" class="thumbnail onebox-avatar" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_500x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_750x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_1000x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_10x10.png">

<h3><a href="https://github.com/mitchcurtis/slate/runs/3115384417" target="_blank" rel="noopener nofollow ugc">Build software better, together</a></h3>

  <p>GitHub is where people build software. More than 65 million people use GitHub to discover, fork, and contribute to over 200 million projects.</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>Where do I go to submit a bug report?</p>

**Comment by airtower-luna** [comment]
<p>Probably <a href="https://github.com/github/docs" class="inline-onebox">GitHub - github/docs: The open-source repo for docs.github.com</a> or <a href="https://github.com/actions/runner" class="inline-onebox">GitHub - actions/runner: The Runner for GitHub Actions</a>, depending on whether you consider it a bug in the documentation or the runner implementation. :wink:</p>

**Comment by mitchcurtis** [comment]
<p>Created <a href="https://github.com/actions/runner/issues/1212" class="inline-onebox" rel="noopener nofollow ugc">pipefail not set by default · Issue #1212 · actions/runner · GitHub</a>.</p>
